llvm/lib/Target/AMDGPU/SIFoldOperands.cpp

```diff
@@ -1793,6 +1793,12 @@ bool SIFoldOperandsImpl::tryFoldOMod(MachineInstr &MI) {
 
   DefOMod->setImm(OMod);
   MRI->replaceRegWith(MI.getOperand(0).getReg(), Def->getOperand(0).getReg());
+  if (Def->getParent() != MI.getParent()) {
```

Comment:
You can just unconditionally clear the flags, they'll just get recomputed again later anyway